hack/switch/switch.sh

```diff
@@ -1,591 +1,60 @@
-                      shift
-                      ;;
-                  gardener)
-                      GARDENER=$1
-                      GARDENER_ARGUMENT=$2
-                      shift
-                      ;;
-                  alias)
-                      ALIAS=$1
-                      ALIAS_ARGUMENTS=$2
-                      ALIAS_ARGUMENTS_ALIAS=$3
-                      shift
-                      ;;
-                  --state-directory)
-                      shift
-                      STATE_DIRECTORY=$1
-                      shift
-                      ;;
-                  --debug)
-                     DEBUG=$1
-                     shift
-                     ;;
-                  --no-index)
-                     NO_INDEX=$1
-                     shift
-                     ;;
-                  --config-path)
-                      shift
-                      CONFIG_PATH=$1
-                      shift
-                      ;;
-                  --hook-name)
-                      shift
-                      HOOK_NAME=$1
-                      shift
-                      ;;
-                  --run-hooks-immediately)
-                      shift
-                      RUN_IMMEDIATELY=$1
-                      shift
-                      ;;
-                  --help)
-                     usage $HOOKS $ALIAS $GARDENER
-                     return
-                     ;;
-                  -h)
-                     usage $HOOKS $ALIAS $GARDENER
-                     return
-                     ;;
-                  version)
-                     VERSION=$1
-                     shift
-                     ;;
-                  -v)
-                     VERSION=$1
-                     shift
-                     ;;
-                  *)
-                     SET_CONTEXT=$1
-                     shift
-                     ;;
-            esac
-    done
-
-
-  if [ -n "$UNSET_CURRENT_CONTEXT" ]
-  then
-     kubectl config unset current-context
-     return
-  fi
-
-  if [ -n "$DELETE_CONTEXT" ]
-  then
-     case $DELETE_CONTEXT in
-       .)
-         kubectl config delete-context $(kubectl config current-context)
-         ;;
-       *)
-         kubectl config delete-context $DELETE_CONTEXT
-         ;;
-       esac
-       return
-  fi
-
-  if [ -n "$CURRENT_CONTEXT" ]
-  then
-     kubectl config current-context
-     return
-  fi
+    case "$1" in
+    --executable-path)
+        EXECUTABLE_PATH="$1"
+        ;;
+    completion)
+        opts+=("$1" --cmd switch)
+        ;;
+    *)
+        opts+=( "$1" )
+        ;;
+    esac
+    shift
+  done
 
   if [ -z "$EXECUTABLE_PATH" ]
   then
-     EXECUTABLE_PATH=$DEFAULT_EXECUTABLE_PATH
-  fi
-
-  if [ -n "$VERSION" ]
-  then
-    $EXECUTABLE_PATH version
-    return
+    EXECUTABLE_PATH="$DEFAULT_EXECUTABLE_PATH"
   fi
 
-  DEBUG_FLAG=''
-  if [ -n "$DEBUG" ]
+  RESPONSE="$($EXECUTABLE_PATH "${opts[@]}")"
+  if [ $? -ne 0 -o -z "$RESPONSE" ]
```

Comment:
I would have to check if all commands still work. But if it does it is a really nice refactoring